unstable/wlr-virtual-pointer-unstable-v1.xml

```diff
@@ -124,5 +131,19 @@
       <arg name="seat" type="object" interface="wl_seat" allow-null="true"/>
       <arg name="id" type="new_id" interface="zwlr_virtual_pointer_v1"/>
     </request>
+    <request name="destroy" type="destructor" since="1">
+      <description summary="destroy the virtual pointer manager"/>
+    </request>
+
+    <request name="create_virtual_pointer2" since="2">
+      <description summary="Create a new virtual pointer">
+        Creates a new virtual pointer. The seat and the output arguments are
+        optional. If they are set, the compositor must bind the pointer
```

Comment:
-1 to "must". These are hints the compositor is free to ignore.

Reply:
So I write software that _maybe_ works based on the whims of the compositor?

Reply:
Most of the Wayland protocols follow the RFC 2119 terms, so "should" would be a good replacement word.

>So I write software that maybe works based on the whims of the compositor?

Wayland in a nutshell.

---

As someone who knows very little about this protocol and don't have a lot of context for it, I find this description extremely vague.

Reply:
>>So I write software that maybe works based on the whims of the compositor?

>Wayland in a nutshell.

It can't be all that bad. In any case I'd like to make things better in that regard, not worse. If the desired behaviour is not guaranteed, there needs to be a way for the client to discover that the compositor decided not to follow its advice so that it can take the appropriate action to avert disaster. I think it's much easier to put strong guarantees into the standard than to add a result event that tells the client about what actually happened.

Reply:
For what it's worth, it's possible for the protocol to place constraints on the compositor's behavior. The compositor doesn't need to follow them - and really, neither does the client - and they can accept the technical and social fallout which ensues. Consider for example that fullscreen clients are requried not to be compositied, which has been a point of contention for us on sway for a while - the more user-friendly option may be to not composite them, but the protocol says so therefore it's right has been the party line for a while (yeah, yeah, there's more nuance to that example, but it's just an example).

Reply:
I changed the wording to "should". Will that suffice?

Reply:
Yes, the compositor will always have the last word. The compositor could already ignore requests sent via virtual-pointer, e.g. if the client isn't authorized to inject events or if the compositor offers a way for users to enable/disable event injection.

The "should" wording LGTM.

One other way to fix this would be to add a "failed" event to the pointer interface to let clients know that the virtual pointer is no longer active. That way, compositors could send the event if  they refuse to create the virtual pointer, if the user has decided to shut down the virtual pointer, or if the compositor refuses to map the pointer to the specified output.

Usually, we stick with the "should" wording so that the compositor can override the client's hints. In this particular case, I don't have a good use-case in mind: not mapping the virtual pointer to the specified output will just result in unexpected behavior. So I'd be fine with the other approach.

Reply:
So, the change is good enough as is?

Reply:
Yes.